lib/eslint/eslint-helpers.js

```diff
@@ -97,6 +105,120 @@ function isGlobPattern(pattern) {
     return isGlob(path.sep === "\\" ? normalizeToPosix(pattern) : pattern);
 }
 
+/**
+ * Searches a directory looking for matching glob patterns. This uses
+ * the config array's logic to determine if a directory or file should
+ * be ignored, so it is consistent with how ignoring works throughout
+ * ESLint.
+ * @param {Object} options The options for this function.
+ * @param {string} options.cwd The directory to search.
+ * @param {Array<string>} options.patterns An array of glob patterns
+ *      to match.
+ * @param {FlatConfigArray} options.configs The config array to use for
+ *      determining what to ignore.
+ * @returns {Promise<Array<string>>} An array of matching file paths
+ *      or an empty array if there are no matches.
+ */
+async function globSearch({ cwd, patterns, configs }) {
+
+    if (patterns.length === 0) {
+        return [];
+    }
+
+    const matchers = patterns.map(pattern => {
+        const patternToUse = path.isAbsolute(pattern)
+            ? normalizeToPosix(path.relative(cwd, pattern))
+            : pattern;
+
+        return new minimatch.Minimatch(patternToUse);
+    });
+
+    return (await doFsWalk(cwd, {
```

Comment:
Starting from `cwd` makes it impossible to lint files outside `cwd`.

For example, do `cd lib` and then `npx eslint "../*.js"`.

Expected result is to lint `Makefile.js`, `karma.conf.js`, and other files in the root of the eslint/eslint project. It used to work before this change, and it works with eslintrc.

Actual result:

```
Oops! Something went wrong! :(

ESLint: 8.24.0

No files matching the pattern "../*.js" were found.
Please check for typing mistakes in the pattern.
```

Reply:
Issue https://github.com/eslint/eslint/issues/16413